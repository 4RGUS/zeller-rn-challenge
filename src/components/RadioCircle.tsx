import React from "react";
import { View, StyleSheet } from "react-native";

type RadioCircleProp = {
  selected: boolean;
};

export const RadioCircle = ({ selected }: RadioCircleProp) => {
  return (
    <View testID="outer-circle" style={styles.outerCircle}>
      {selected && <View testID="inner-circle" style={styles.innerCircle} />}
    </View>
  );
};

const styles = StyleSheet.create({
  outerCircle: {
    height: 18,
    width: 18,
    borderRadius: 9,
    borderWidth: 2,
    borderColor: "#4D94FF",
    alignItems: "center",
    justifyContent: "center",
  },
  innerCircle: {
    height: 8,
    width: 8,
    borderRadius: 4,
    backgroundColor: "#4D94FF",
  },
});
