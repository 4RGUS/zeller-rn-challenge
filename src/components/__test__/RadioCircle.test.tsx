import React from "react";
import { render } from "@testing-library/react-native";
import { RadioCircle } from "../RadioCircle";

describe("RadioCircle component", () => {
  it("matches snapshot when not selected", () => {
    const { toJSON } = render(<RadioCircle selected={false} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("matches snapshot when selected", () => {
    const { toJSON } = render(<RadioCircle selected={true} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
