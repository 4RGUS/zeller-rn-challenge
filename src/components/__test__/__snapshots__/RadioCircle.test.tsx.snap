// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioCircle component matches snapshot when not selected 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "#4D94FF",
      "borderRadius": 9,
      "borderWidth": 2,
      "height": 18,
      "justifyContent": "center",
      "width": 18,
    }
  }
/>
`;

exports[`RadioCircle component matches snapshot when selected 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "#4D94FF",
      "borderRadius": 9,
      "borderWidth": 2,
      "height": 18,
      "justifyContent": "center",
      "width": 18,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#4D94FF",
        "borderRadius": 4,
        "height": 8,
        "width": 8,
      }
    }
  />
</View>
`;
